charms/istio-pilot/src/charm.py

```diff
@@ -352,35 +353,62 @@ def _istiod_svc(self):
         else:
             return exporter_ip
 
-    @property
-    def _gateway_address(self):
-        """Look up the load balancer address for the ingress gateway.
-        If the gateway isn't available or doesn't have a load balancer address yet,
-        returns None.
-        """
-        # FIXME: service name is hardcoded
+    def _is_gateway_service_up(self):
+        """Returns True if the gateway is up, else False."""
+        svc = self._get_gateway_service()
+
+        if svc.spec.type == "NodePort":
+            # TODO: do we need to interrogate this further for status?
+            return True
+        else:
+            if _get_gateway_address_from_svc(svc) is not None:
+                return True
+            else:
+                return False
+
+    def _get_gateway_service(self):
```

Comment:
```suggestion
    def _get_gateway_service(self):
        """Returns `Service` lightkube object..."""
```

Reply:
Updated in most recent commit

Reply:
@ca-scribner @DnPlas 
why not add all code suggestions in batch from github itself ?

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

Reply:
yeah I did for a few, but there were a few I rewrote a little.  For example, I think @DnPlas meant this one as a notional suggestion rather than something to be taken verbatim (I don't think she actually wanted us to write `lightkube object...`)